src/protocol/basics/reshare.rs

```diff
@@ -280,6 +346,37 @@ mod tests {
             }
         }
 
+        /// Validates that reshare protocol actually generates new xor shares using PRSS.
+        #[tokio::test]
+        async fn generates_unique_xor_shares() {
+            let world = TestWorld::new().await;
+
+            for &target in Role::all() {
+                let secret = thread_rng().gen::<BitArray40>();
+                let shares = world
+                    .semi_honest(secret, |ctx, share: XorShare<BitArray40>| async move {
```

Comment:
Please add an additional test for `.malicious' and validate that *both* parts (`x` and `rx`) have changed.

Reply:
We don't have `x` and `rx` for XorShare. `MaliciousContext` is generic over `F`, so there's no way to upgrade XorShare.